### PR TITLE
zf3 support

### PR DESCRIPTION
**Pull request by necromant2005**
zf3 support. Unfortunately tests are working only in zf2 or zf3.

### CODE REVIEW

**Comment by necromant2005** [comment]
fixed tests and travis. config

**Comment by neilime** [comment]
@necromant2005 : the phpunit tests are broken, do you you prefer I merge your work in the PR #180. Or do you want to fix it in your PR ? Let me know.

**Comment by necromant2005** [comment]
I'll fix it. The issue somewhere in travis environment because it passes local.

**Comment by necromant2005** [comment]
fixed! https://travis-ci.org/necromant2005/zf2-twb-bundle

**Comment by neilime** [comment]
Thank you.

One last thing, can you update the `composer.json` to set the minimal working version of dependencies. Example : 

```json
        "zendframework/zend-mvc": "~3.0.2",
        "zendframework/zend-form": "~2.9.0",
        "zendframework/zend-mvc-i18n": "~1.0.0"
```

**Comment by exptom** [comment]
Any ETA on when this might be in a tagged release?

File: src/TwbBundle/Options/Factory/ModuleOptionsFactory.php [hunk 1]
```diff
@@ -14,4 +15,11 @@ public function createService(ServiceLocatorInterface $serviceLocator)
         $options = $config['twbbundle'];
         return new ModuleOptions($options);
     }
+
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = NULL)
+    {
+        $config = $container->get('Config');
```
Comment (thomasvargiu) [inline src/TwbBundle/Options/Factory/ModuleOptionsFactory.php]: - The standard is to use lowercase keys.

File: composer.json [hunk 2]
```diff
@@ -22,18 +22,18 @@
...
-        "zendframework/zend-i18n": "~2.3",
-        "zendframework/zend-loader": "~2.3",
-        "zendframework/zend-log": "~2.3",
-        "zendframework/zend-modulemanager": "~2.3",
-        "zendframework/zend-mvc": "~2.3",
-        "zendframework/zend-serializer": "~2.3",
-        "zendframework/zend-servicemanager": "~2.3",
-        "zendframework/zend-stdlib": "~2.3",
-        "zendframework/zend-view": "~2.3"
+        "zendframework/zend-config": "*",
```
Comment (thomasvargiu) [inline composer.json]: Versions should be specified like this:
```
"zendframework/zend-servicemanager": "^2.5 || ^3.0.1"
```
using the `^` comparation.

File: src/TwbBundle/Form/View/Helper/Factory/TwbBundleFormElementFactory.php [hunk 3]
```diff
@@ -19,4 +20,10 @@ public function createService(ServiceLocatorInterface $serviceLocator)
         $options = $serviceLocator->getServiceLocator()->get('TwbBundle\Options\ModuleOptions');
         return new TwbBundleFormElement($options);
     }
+
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = NULL)
```
Comment (thomasvargiu) [inline src/TwbBundle/Form/View/Helper/Factory/TwbBundleFormElementFactory.php]: For every factory:

- `createService()` method should delegate `__invoke()` method for create services, without duplicate code.
- `null` keywork must be lowercase.

Then the `__invoke()` method should use the main container like this:

```
public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
{
    if ($container instanceof AbstractPluginManager) {
        $container = $container->getServiceLocator() ?: $container;
    }

    // ....
}
```

**Comment by thomasvargiu** [comment]
@necromant2005 Can you check my notes and update the code?

**Comment by neilime** [comment]
I'm waiting for the last updates from @necromant2005 before merging this great pull request : 

- Updating the composer.json to set the minimal working version of dependencies
- Notes from @thomasvargiu 

Thanks

**Comment by exptom** [comment]
I'm happy to make these changes do I just branch this PR and create a new one?

**Comment by exptom** [comment]
@neilime I've done these changes in #181. Would it be possible to get a new tagged version?

**Comment by neilime** [comment]
@exptom : done ! Thank you for your work

**Pull request merged**